deployer/src/runtime_manager.rs

```diff
@@ -115,19 +131,56 @@ impl RuntimeManager {
         .await
         .context("failed to start shuttle runtime")?;
 
+        let process = Arc::new(Mutex::new(process));
+        let process_cloned = Arc::clone(&process);
         self.runtimes
             .lock()
             .unwrap()
             .insert(id, (process, runtime_client.clone()));
 
+        let stdout = process_cloned
+            .lock()
+            .await
+            .stdout
+            .take()
+            .context("child process did not have a handle to stdout")?;
+
+        // Ensure the child process is spawned in the runtime so it can
+        // make progress on its own while we await for any output.
+        tokio::spawn(async move {
+            process_cloned
+                .lock()
+                .await
+                .wait()
+                .await
+                .expect("child process encountered an error");
+        });
+
+        let mut reader = BufReader::new(stdout).lines();
+        let mut logger_client = self.logger_client.clone();
+        tokio::spawn(async move {
+            while let Some(line) = reader.next_line().await.unwrap() {
+                // TODO: `store_logs` accepts a Vec but logs are sent one by one. Is this feasible?
```

Comment:
I think we can probably start by adding some default batching logic here. It should account lets say for batching 100 logs one time, but it should also account for some time window. E.g. whenever deciding if we want to store the logs, we would verify the batch size against the default value or check whether the window has passed.

Reply:
Agreed. Resolving this.